lib/semantics/check-do.cc

```diff
@@ -167,15 +167,10 @@ class DoConcurrentBodyEnforce {
         }
       }
     } else {
-      // C1139: check for an impure procedure component
-      auto &component{std::get<parser::ProcComponentRef>(procedureDesignator.u)
-                          .v.thing.component};
-      if (component.symbol && !IsPureProcedure(*component.symbol)) {
-        SayWithDo(context_, currentStatementSourcePosition_,
-            "Call to an impure procedure component is not allowed"
-            " in DO CONCURRENT"_err_en_US,
-            doConcurrentSourcePosition_);
-      }
+      // C1139: this a procedure component, all of which are impure
```

Comment:
I do not believe that the statement in the comment is correct.  Procedure components can have specific interfaces, those interfaces imply characteristics, and one of those characteristics is whether or not the procedure is `PURE`.

Reply:
Thanks, Peter.  I hadn't previously noticed the ability to specify PURE procedure interfaces.  I'll fix the test and the code to check for this.